llvm/test/Transforms/SLPVectorizer/X86/pr98978.ll

```diff
@@ -0,0 +1,127 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S -passes=slp-vectorizer < %s | FileCheck %s
```

Comment:
Should we also have a BasicAA analysis test based on this?

Reply:
The BasicAA analysis tests don't use BatchAA. I think it might be possible to construct a case where this issue also occurs without BatchAA, but it's not easy.

Reply:
Ah that's a shame, relying on SLP seems a bit fragile, but it sounds like it's the best option we have currently.